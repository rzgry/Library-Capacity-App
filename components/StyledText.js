import React from 'react';
import { Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  heading: { fontWeight: '800', fontSize: 24 },
  subheading: { fontWeight: '800', fontSize: 20 },
});

export const MonoText = ({ style, ...rest }) => (
  <Text {...rest} style={[style, { fontFamily: 'space-mono' }]} />
);

export const Heading = ({ style, ...rest }) => <Text {...rest} style={[style, styles.heading]} />;

export const SubHeading = ({ style, ...rest }) => (
  <Text {...rest} style={[style, styles.subheading]} />
);
