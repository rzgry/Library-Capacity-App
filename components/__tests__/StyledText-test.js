import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { MonoText, Heading, SubHeading } from '../StyledText';

describe('<MonoText />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Heading />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Heading>Snapshot test!</Heading>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<Subheading />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SubHeading>Snapshot test!</SubHeading>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
