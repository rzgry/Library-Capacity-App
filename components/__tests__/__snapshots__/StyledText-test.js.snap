// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontSize": 24,
        "fontWeight": "800",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`<MonoText /> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`<Subheading /> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontSize": 20,
        "fontWeight": "800",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
